auth/interface.go

```diff
@@ -30,3 +39,14 @@ type Minter interface {
 	// VerifyLSAT attempts to verify an LSAT with the given parameters.
 	VerifyLSAT(context.Context, *mint.VerificationParams) error
 }
+
+// InvoiceChecker is an entity that is able to check the status of an invoice,
+// particularly whether it's been paid or not.
+type InvoiceChecker interface {
+	// VerifyInvoiceStatus checks that an invoice identified by a payment
```

Comment:
Given the control flow above, a user will only enter this verification phase once they have the pre-image, correct? So at this point there shouldn't be any risk of the user gaining the pre-image before we settle things on our end.

Reply:
The whole timeout thing was added to make sure the call from the client (that includes the preimage) doesn't arrive before we get the update from our invoices subscription gRPC stream, in case `lnd` is slow to send it. Or what exactly do you mean?